src/helpers/encryption.ts

```diff
@@ -1,123 +1,107 @@
+ * @param privatePGPKey     Private Key to decrypt the message.
+ * @see                     Learn more about encryption {@link https://docs.openpgpjs.org/global.html#decrypt here}.
+ * @returns                 A base64 string containing the decrypted message.
+ */
+export const decryptFile = async (encryptedMessage: string, privatePGPKey: string) => {
+  const privateKey = await openpgp.readPrivateKey({ armoredKey: privatePGPKey })
   const message = await openpgp.readMessage({
     armoredMessage: encryptedMessage,
   })
-
-  const { data: decrypted } = await openpgp.decrypt({
+  const { data: decryptedMessage } = await openpgp.decrypt({
     message,
     decryptionKeys: privateKey,
   })
 
-  return decrypted
+  return decryptedMessage
 }
 
-
 /**
- * @name encryptAndUploadFile
- * @summary             Encrypts and uploads a file on an IFPS gateway.
- * @param file          File to encrypt and then upload on IPFS.
- * @param ipfsGateway   IPFS gateway to upload your file on. Default is https://ipfs.ternoa.dev/api/v0/add
- * @param apiKey        API Key to validate the upload on the IPFS gateway.
- * @returns             An array containing both secretFile IPFS hash and publicPGPKey hash.
+ * @name secretNftEncryptAndUploadFile
+ * @summary                 Encrypts and uploads a file on an IFPS gateway.
+ * @param file              File to encrypt and then upload on IPFS.
+ * @param publicPGPKey      Public Key to encrypt the file.
+ * @param ipfsGateway       IPFS gateway to upload your file on. Default is https://ipfs.ternoa.dev/api/v0/add
+ * @param apiKey            API Key to validate the upload on the IPFS gateway.
+ * @param metadata          Optional secret NFT metadata (Title, Description) {@link https://github.com/capsule-corp-ternoa/ternoa-proposals/blob/main/TIPs/tip-510-Secret-nft.md here}.
+ * @returns                 The data object with the secret NFT IPFS hash (ex: to add as offchain secret metadatas in the extrinsic).
  */
-export const encryptAndUploadFile = async (
+export const secretNftEncryptAndUploadFile = async <T>(
+  file: File,
   publicPGPKey: string,
-  file: Buffer,
-  ipfsGateway?: string,
+  ipfsGateway?: URL,
   apiKey?: string,
+  metadata?: SecretNftMetadataType<T>,
 ) => {
-  // const encryptedFile = await encryptFile(file, publicPGPKey)
-  // const encryptedFileDataBuffer = Buffer.from(encryptedFile as string)
-  // const publicPGPKeyBuffer = Buffer.from(publicPGPKey as string)
-  // const [secretIpfsUploadRes, publicPGPKeyIpfsUploadRes] = await Promise.all([
-  //   ipfsFileUpload(encryptedFileDataBuffer, ipfsGateway, apiKey),
-  //   ipfsFileUpload(publicPGPKeyBuffer, ipfsGateway, apiKey),
-  // ])
-  // return [secretIpfsUploadRes, publicPGPKeyIpfsUploadRes]
-}
+  if (!file) throw new Error(`${Errors.IPFS_FILE_UPLOAD_ERROR} - File undefined`)
+  const encryptedFile = (await encryptFile(file, publicPGPKey)) as string
+  const ipfsClient = new TernoaIPFS(ipfsGateway, apiKey)
```

Comment:
IPFS client init should not be handle inside this helper. We should passed the client instance as parameter:

```typescript

export const secretNftEncryptAndUploadFile = async <T>(
  file: File,
  publicPGPKey: string,
  ipfsClient: TernoaIPFS,
  metadata?: SecretNftMetadataType<T>,
) => {
  if (!file) throw new Error(`${Errors.IPFS_FILE_UPLOAD_ERROR} - File undefined`)
  const encryptedFile = (await encryptFile(file, publicPGPKey)) as string
  const ipfsRes = await ipfsClient.storeSecretNFT(encryptedFile, publicPGPKey, metadata, file.type)

  return ipfsRes
}
```